test/Concurrency/async_task_groups_and_type_inference.swift

```diff
@@ -0,0 +1,99 @@
+// RUN: %target-swift-frontend  -disable-availability-checking %s -emit-sil -o /dev/null -verify -strict-concurrency=complete
+
+// REQUIRES: concurrency
+// REQUIRES: asserts
+
+// === OK cases
+
+let _ = await withTaskGroup { group in
+  group.addTask { }
+  group.addTask { }
+}
+let _ = await withThrowingTaskGroup { group in
+  group.addTask { }
+  group.addTask { }
+}
+
+let _ = await withTaskGroup { group in
+  group.addTask { 1 }
+  group.addTask { 2 }
+  return "result"
+}
```

Comment:
@ktoso, should I also add a similar test as this for `withThrowingTaskGroup`?

Reply:
It wouldn't hurt, if you want to. Always good to cover all the APIs yes

Reply:
I added the mirrored `withThrowingTaskGroup` test and also added a pair of tests to ensure that multi-statement closure inference works as expected, i.e.:

```swift
let _ = await withTaskGroup { group in
  group.addTask {
    if Bool.random() {
      return 1
    } else {
      return 0
    }
  }
  group.addTask {
    if Bool.random() {
      return 1
    } else {
      return 0
    }
  }
  return "result"
}
```

Reply:
Thanks!